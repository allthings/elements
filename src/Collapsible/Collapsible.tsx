import React, { Component, KeyboardEvent } from 'react'
import { css } from 'glamor'
import Icon from '../Icon'
import Text from '../Text'
import View from '../View'
import { ColorPalette, alpha } from '@allthings/colors'

const tick = () => new Promise(resolve => setTimeout(resolve, 0))

interface ICollapsibleProps {
  readonly title?: string
  readonly initiallyCollapsed?: boolean
  readonly hasBottomBorder?: boolean
  readonly tabIndex?: number
  readonly onToggle?: (on: boolean) => void
}
interface IState {}

/**
 * A Collapsible is a simple container, that makes it possible to change between collapsed and extended states, and this way hiding and showing the children passed in.
 * ```js
 * <ThemeProvider>
 *  <Card>
 *    <Collapsible
 *      title="Address"
 *      hasBottomBorder
 *      initiallyCollapsed={false}
 *      tabIndex={1}
 *    >
 *      <CardContent>
 *        <Text>Kaiser Joseph Str. 260</Text>
 *      </CardContent>
 *    </Collapsible>
 *    <Collapsible
 *        title="Contact"
 *        initiallyCollapsed={true}
 *        tabIndex={2}
 *    >
 *      <CardContent>
 *        <Text>1(23) 456-7890</Text>
 *      </CardContent>
 *    </Collapsible>
 *  </Card>
 * </ThemeProvider>
 * ```
 **/
class Collapsible extends Component<ICollapsibleProps, IState> {
  static defaultProps = {
    initiallyCollapsed: true,
    hasBottomBorder: false,
  }

  state = {
    collapsed: this.props.initiallyCollapsed,
    overflow: this.props.initiallyCollapsed ? 'hidden' : null,
  }

  async componentDidMount() {
    const { current } = this.childRef

    if (current && current.style) {
      if (!this.props.initiallyCollapsed) {
        current.style.height = `${current.scrollHeight}px`
        await tick()
        current.style.height = 'auto'
      } else {
        current.style.height = `0px`
      }
    }
  }

  childRef = React.createRef<HTMLDivElement>()

  toggleCollapse = async () => {
    const { current } = this.childRef
    if (current && current.style) {
      if (current.style.height !== '0px') {
        current.style.height = `${current.scrollHeight}px`
        await tick()
        current.style.height = '0px'
        this.setState({ collapsed: true, overflow: 'hidden' })
        // signal new state for the parent
        this.props.onToggle && this.props.onToggle(true)
      } else {
        current.style.height = `${current.scrollHeight}px`
        this.setState({ collapsed: false })
        // signal new state for the parent
        this.props.onToggle && this.props.onToggle(false)
      }
    }
  }

  handleTransitionEnd = () => {
    if (
      !this.state.collapsed &&
      this.childRef.current &&
      this.childRef.current.style
    ) {
      this.childRef.current.style.height = 'auto'
      this.setState({ overflow: null })
    }
  }

  onKeyPress = (e: KeyboardEvent) => e.key === 'Enter' && this.toggleCollapse()

  render() {
    const { title, children, hasBottomBorder, tabIndex } = this.props

    const { collapsed, overflow } = this.state
    return (
      <View
        direction="column"
        {...css({
          borderBottom:
            hasBottomBorder && `1px solid ${ColorPalette.lightGrey}`,
          width: '100%',
        })}
      >
        <View
          direction="row"
          alignH="space-between"
          onClick={this.toggleCollapse}
          {...css({
            backgroundColor: 'white',
            cursor: 'pointer',
          })}
        >
          <View
            direction="row"
            alignV="center"
            {...css({
              height: 50,
              padding: '0 20px',
              width: 'calc(100% - 50px)',
            })}
          >
            <Text
              strong
              style={{
                width: '100%',
                whiteSpace: 'nowrap',
                overflow: 'hidden',
                textOverflow: 'ellipsis',
              }}
            >
              {title}
            </Text>
          </View>
          {/* arrow */}
          <View
            direction="row"
            alignV="center"
            alignH="center"
            {...css({
              width: '50px',
              height: '50px',
            })}
          >
            <View
              {...css({
                padding: '3px 5px 8px 5px',
                borderRadius: '2px',
                ':focus': { border: `2px solid ${alpha('#59baf7', 0.5)}` },
                outline: 'none',
              })}
              onKeyPress={this.onKeyPress}
              tabIndex={tabIndex}
            >
              <Icon
                name={collapsed ? 'arrow-down-filled' : 'arrow-up-filled'}
                size={10}
                color="lightGreyIntense"
              />
            </View>
          </View>
        </View>
        <View
          ref={this.childRef}
          onTransitionEnd={this.handleTransitionEnd}
          {...css({
            transitionProperty: 'height',
            transition: 'height 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms',
            overflow,
            transformOrigin: 'top',
          })}
        >
          {children}
        </View>
      </View>
    )
  }
}

export default Collapsible
