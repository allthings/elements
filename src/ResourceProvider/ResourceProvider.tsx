import * as React from 'react'

export const RESOURCE_PATH = 'https://static.allthings.me'

/**
 * Elements uses a set of static resources like images or icons.
 * In order to benefit from caching across all apps, these resources are provided by a static asset CDN.
 *
 * Whenever you like to use Icons or Illustrations, you need to use the ResourceProvider to let the components know where they are.
 */
export const ResourceProviderContext = React.createContext({
  resourcePath: RESOURCE_PATH,
})

export interface IResourceProviderProps {
  readonly resourcePath?: string
}

/**
 * All elements support theming by default, and therefore every element must be wrapped inside a ThemeProvider.
 * The ThemeProvider allows you to define the default colors for most elements.
 *
 * **Example**: If you want all you buttons to be red, instead of writing <Button color="red" /> all the time, you might want to set the "primary" color of your theme to red.
 **/
const ResourceProvider: React.FC<IResourceProviderProps> = ({
  children,
  resourcePath = RESOURCE_PATH,
}) => {
  // this is to prevent unintentional re-renders in consumers
  // see https://reactjs.org/docs/context.html#caveats
  const [value, setValue] = React.useState({ resourcePath })
  React.useEffect(() => {
    setValue({ resourcePath })
  }, [resourcePath])

  return (
    <ResourceProviderContext.Provider value={value}>
      {children}
    </ResourceProviderContext.Provider>
  )
}

export default ResourceProvider
