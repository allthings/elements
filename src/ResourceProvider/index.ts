export { default } from './ResourceProvider'
export * from './ResourceProvider'
