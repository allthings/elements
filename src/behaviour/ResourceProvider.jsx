import React from 'react'
import PropTypes from 'prop-types'

export const RESOURCE_PATH = 'https://static.allthings.me'

/**
 * Elements uses a set of static resources like images or icons.
 * In order to benefit from caching across all apps, these resources are provided by a static asset CDN.
 *
 * Whenever you like to use Icons or Illustrations, you need to use the ResouceProvider to let the components know where they are.
 */
class ResourceProvider extends React.Component {
  getChildContext() {
    return { resourcePath: this.props.resourcePath }
  }

  static childContextTypes = {
    resourcePath: PropTypes.string,
  }

  static propTypes = {
    resourcePath: PropTypes.string.isRequired,
    children: PropTypes.node.isRequired,
  }

  static defaultProps = {
    resourcePath: RESOURCE_PATH,
    children: null,
  }

  render() {
    return this.props.children
  }
}

export default ResourceProvider
