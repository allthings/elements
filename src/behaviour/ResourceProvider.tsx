import React from 'react'

export const RESOURCE_PATH = 'https://static.allthings.me'

/**
 * Elements uses a set of static resources like images or icons.
 * In order to benefit from caching across all apps, these resources are provided by a static asset CDN.
 *
 * Whenever you like to use Icons or Illustrations, you need to use the ResouceProvider to let the components know where they are.
 */
const ResouceProvider = React.createContext({
  resourcePath: RESOURCE_PATH,
})

export type ResourceType = {
  resourcePath?: string
}

export default ResouceProvider
