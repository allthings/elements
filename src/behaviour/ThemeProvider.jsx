import React from 'react'
import PropTypes from 'prop-types'
import { Broadcast, Subscriber } from 'react-broadcast'
import { ColorPalette } from '@allthings/colors'

export const THEME_CHANNEL = 'theme'

export const defaultTheme = {
  primary: ColorPalette.primary.blue,
  text: ColorPalette.text.primary,
  secondaryText: ColorPalette.text.primary,
  titleColor: ColorPalette.text.primary,
  contrast: ColorPalette.white,
  warn: ColorPalette.red,
  disabled: ColorPalette.grey,
  background: ColorPalette.background.bright,
  textOnBackground: ColorPalette.white,
}

/**
* All elements support theming by default, and therefore every element must be wrapped inside a ThemeProvider.
* The ThemeProvider allows you to define the default colors for most elements.
*
* **Example**: If you want all you buttons to be red, instead of writing <Button color="red" /> all the time, you might want to set the "primary" color of your theme to red.
**/
export default class ThemeProvider extends React.PureComponent {
  static propTypes = {
    root: PropTypes.bool,
    theme: PropTypes.shape({
      primary: PropTypes.string,
      text: PropTypes.string,
      secondaryText: PropTypes.string,
      titleColor: PropTypes.string,
      contrast: PropTypes.string,
      warn: PropTypes.string,
      disabled: PropTypes.string,
      background: PropTypes.string,
      textOnBackground: PropTypes.string,
    }),
    children: PropTypes.node,
  }

  static defaultProps = {
    theme: defaultTheme,
  }

  static contextTypes = Subscriber.contextTypes

  render() {
    let root = true
    if (this.context.broadcasts && this.context.broadcasts[THEME_CHANNEL]) {
      root = false
    }
    const { theme, children } = this.props
    return root ? (
      <Broadcast
        channel={THEME_CHANNEL}
        value={{
          ...ThemeProvider.defaultProps.theme,
          ...theme,
        }}
      >
        {children}
      </Broadcast>
    ) : (
      <Subscriber channel={THEME_CHANNEL}>
        {previousTheme => (
          <Broadcast
            channel={THEME_CHANNEL}
            value={{
              ...ThemeProvider.defaultProps.theme,
              ...previousTheme,
              ...theme,
            }}
          >
            {children}
          </Broadcast>
        )}
      </Subscriber>
    )
  }
}

export const ThemeShape = ThemeProvider.propTypes.theme

export const withTheme = (mapThemeToProps, displayName) => WrappedComponent =>
  // eslint-disable-next-line
  class extends React.PureComponent {
    static displayName = displayName || WrappedComponent.displayName
    static component = WrappedComponent

    renderComponent = theme => {
      const props = !mapThemeToProps
        ? { theme }
        : mapThemeToProps(theme, this.props)

      return <WrappedComponent {...this.props} {...props} />
    }

    render() {
      return (
        <Subscriber channel={THEME_CHANNEL}>{this.renderComponent}</Subscriber>
      )
    }
  }
