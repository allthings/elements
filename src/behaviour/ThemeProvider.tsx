import React, { FunctionComponent, createContext } from 'react'
import { ColorPalette } from '@allthings/colors'

export const defaultTheme = {
  primary: ColorPalette.primary.blue,
  text: ColorPalette.text.primary,
  secondaryText: ColorPalette.text.secondary,
  titleColor: ColorPalette.text.primary,
  contrast: ColorPalette.white,
  warn: ColorPalette.red,
  disabled: ColorPalette.grey,
  background: ColorPalette.background.bright,
  textOnBackground: ColorPalette.white,
}

export interface ITheme {
  primary: string
  text: string
  secondaryText: string
  titleColor: string
  contrast: string
  warn: string
  disabled: string
  background: string
  textOnBackground: string
  [key: string]: string
}

const ThemeContext = createContext<ITheme>(defaultTheme)
export const ThemeConsumer = ThemeContext.Consumer

/**
 * All elements support theming by default, and therefore every element must be wrapped inside a ThemeProvider.
 * The ThemeProvider allows you to define the default colors for most elements.
 *
 * **Example**: If you want all you buttons to be red, instead of writing <Button color="red" /> all the time, you might want to set the "primary" color of your theme to red.
 **/
const ThemeProvider: FunctionComponent<{ theme?: Partial<ITheme> }> = ({
  children,
  theme = {},
}) => (
  <ThemeConsumer>
    {contextTheme => (
      <ThemeContext.Provider value={{ ...contextTheme, ...(theme as ITheme) }}>
        {children}
      </ThemeContext.Provider>
    )}
  </ThemeConsumer>
)

export default ThemeProvider

type mapThemeToPropsType = (theme: ITheme, props: any) => any

function createThemeConsumer<T>(
  mapThemeToProps: mapThemeToPropsType,
  WrappedComponent: React.ComponentType<T>,
  props: Omit<T, 'theme'>
) {
  return (theme: ITheme) => {
    const extraProps = !mapThemeToProps
      ? { theme }
      : mapThemeToProps(theme, props)

    return <WrappedComponent {...props} {...extraProps} />
  }
}

export function withTheme(mapThemeToProps: mapThemeToPropsType) {
  return function<T extends { theme: ITheme }>(
    WrappedComponent: React.ComponentType<T>
  ): React.ComponentType<Omit<T, 'theme'>> {
    return function(props: Omit<T, 'theme'>) {
      return (
        <ThemeConsumer>
          {createThemeConsumer<T>(mapThemeToProps, WrappedComponent, props)}
        </ThemeConsumer>
      )
    }
  }
}
