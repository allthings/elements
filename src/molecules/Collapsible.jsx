import PropTypes from 'prop-types'
import React from 'react'
import { css } from 'glamor'
import { ColorPalette, alpha } from '@allthings/colors'
import Icon from '../atoms/Icon'
import View from '../atoms/View'
import Text from '../atoms/Text'

const tick = () => new Promise(resolve => setTimeout(resolve, 0))

/**
 * A Collapsible is a simple container, that makes it possible to change between collapsed and extended states, and this way hiding and showing the children passed in.
 * ```js
 * <ThemeProvider>
 *  <Card>
 *    <Collapsible
 *      title="Address"
 *      hasBottomBorder
 *      initiallyCollapsed={false}
 *      tabIndex={1}
 *    >
 *      <CardContent>
 *        <Text>Kaiser Joseph Str. 260</Text>
 *      </CardContent>
 *    </Collapsible>
 *    <Collapsible
 *        title="Contact"
 *        initiallyCollapsed={true}
 *        tabIndex={2}
 *    >
 *      <CardContent>
 *        <Text>1(23) 456-7890</Text>
 *      </CardContent>
 *    </Collapsible>
 *  </Card>
 * </ThemeProvider>
 * ```
 **/
class Collapsible extends React.Component {
  static propTypes = {
    title: PropTypes.string,
    children: PropTypes.node,
    initiallyCollapsed: PropTypes.bool,
    hasBottomBorder: PropTypes.bool,
    tabIndex: PropTypes.number,
    onToggle: PropTypes.func,
  }

  static defaultProps = {
    initiallyCollapsed: true,
    hasBottomBorder: false,
    tabIndex: null,
    onToggle: () => {},
  }

  state = {
    collapsed: this.props.initiallyCollapsed,
    overflow: this.props.initiallyCollapsed ? 'hidden' : null,
  }

  async componentDidMount() {
    const { current } = this.childRef

    if (current) {
      if (!this.props.initiallyCollapsed) {
        current.style.height = `${current.scrollHeight}px`
        await tick()
        current.style.height = 'auto'
      } else {
        current.style.height = `0px`
      }
    }
  }

  childRef = React.createRef()

  toggleCollapse = async () => {
    const { current } = this.childRef
    if (current.style.height !== '0px') {
      current.style.height = `${current.scrollHeight}px`
      await tick()
      current.style.height = '0px'
      this.setState({ collapsed: true, overflow: 'hidden' })
      // signal new state for the parent
      this.props.onToggle(true)
    } else {
      current.style.height = `${current.scrollHeight}px`
      this.setState({ collapsed: false })
      // signal new state for the parent
      this.props.onToggle(false)
    }
  }

  handleTransitionEnd = () => {
    if (!this.state.collapsed) {
      this.childRef.current.style.height = 'auto'
      this.setState({ overflow: null })
    }
  }

  onKeyPress = e => e.key === 'Enter' && this.toggleCollapse()

  render() {
    const { title, children, hasBottomBorder, tabIndex } = this.props

    const { collapsed, overflow } = this.state
    return (
      <View
        direction="column"
        {...css({
          borderBottom:
            hasBottomBorder && `1px solid ${ColorPalette.lightGrey}`,
          width: '100%',
        })}
      >
        {/* Header bar */}
        <View
          direction="row"
          alignH="space-between"
          onClick={this.toggleCollapse}
          {...css({
            backgroundColor: 'white',
            cursor: 'pointer',
          })}
        >
          {/* Title */}
          <View
            direction="row"
            alignV="center"
            {...css({
              height: 50,
              padding: '0 20px',
              width: 'calc(100% - 50px)',
            })}
          >
            <Text
              strong
              style={{
                width: '100%',
                whiteSpace: 'nowrap',
                overflow: 'hidden',
                textOverflow: 'ellipsis',
              }}
            >
              {title}
            </Text>
          </View>
          {/* arrow */}
          <View
            direction="row"
            alignV="center"
            alignH="center"
            {...css({
              width: '50px',
              height: '50px',
            })}
          >
            <View
              {...css({
                padding: '3px 5px 8px 5px',
                borderRadius: '2px',
                ':focus': { border: `2px solid ${alpha('#59baf7', 0.5)}` },
                outline: 'none',
              })}
              onKeyPress={this.onKeyPress}
              tabIndex={tabIndex}
            >
              <Icon
                name={collapsed ? 'arrow-down-filled' : 'arrow-up-filled'}
                size={10}
                color="lightGreyIntense"
              />
            </View>
          </View>
        </View>
        {/* Child */}
        <View
          onRef={this.childRef}
          onTransitionEnd={this.handleTransitionEnd}
          {...css({
            transitionProperty: 'height',
            transition: 'height 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms',
            overflow,
            transformOrigin: 'top',
          })}
        >
          {children}
        </View>
      </View>
    )
  }
}

export default Collapsible
