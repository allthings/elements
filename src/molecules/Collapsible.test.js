import React from 'react'
import renderer from 'react-test-renderer'
import ThemeProvider from '../behaviour/ThemeProvider'
import ResourceProvider from '../behaviour/ResourceProvider'
import Collapsible from '../molecules/Collapsible'
import genProps from 'react-generate-props'

describe('Input component', () => {
  test('Collapsible renders properly', () => {
    const props = genProps(Collapsible, { optional: true })
    const tree = renderer
      .create(
        <ThemeProvider>
          <ResourceProvider>
            <Collapsible {...props}>
              <p>Test child</p>
            </Collapsible>
          </ResourceProvider>
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
