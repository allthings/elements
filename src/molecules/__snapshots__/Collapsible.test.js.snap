// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Collapsible renders properly 1`] = `
.css-tziwal,
[data-css-tziwal] {
  border-bottom: 1px solid #ecf0f1;
  width: 100%;
}

.css-4bx01r,
[data-css-4bx01r] {
  background-color: white;
  cursor: pointer;
}

.css-1eav4r6,
[data-css-1eav4r6] {
  height: 50px;
  padding: 0 20px;
  width: calc(100% - 50px);
}

.css-1di7qz,
[data-css-1di7qz] {
  width: 50px;
  height: 50px;
}

.css-1l7pri3,
[data-css-1l7pri3] {
  padding: 3px 5px 8px 5px;
  border-radius: 2px;
  outline: none;
}

.css-1l7pri3:focus,
[data-css-1l7pri3]:focus,
.css-1l7pri3[data-simulate-focus],
[data-css-1l7pri3][data-simulate-focus] {
  border: 2px solid rgba(89, 186, 247, 0.5);
}

.css-dbct2k,
[data-css-dbct2k] {
  transition-property: height;
  transition: height 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  overflow: hidden;
  transform-origin: top;
  -webkit-transition-property: height;
  -moz-transition-property: height;
  -webkit-transition: height 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  -moz-transition: height 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  -webkit-transform-origin: top;
}

.css-iar4wy,
[data-css-iar4wy] {
  box-sizing: border-box;
  align-content: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  flex: 0 0 auto;
  -webkit-align-content: stretch;
  -webkit-box-align: stretch;
  -webkit-align-items: stretch;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -webkit-flex: 0 0 auto;
}

.css-f86ip,
[data-css-f86ip] {
  box-sizing: border-box;
  align-content: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  cursor: pointer;
  flex: 0 0 auto;
  -webkit-align-content: stretch;
  -webkit-box-align: stretch;
  -webkit-align-items: stretch;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -webkit-flex: 0 0 auto;
}

.css-h7582n,
[data-css-h7582n] {
  box-sizing: border-box;
  align-content: center;
  align-items: center;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  flex: 0 0 auto;
  -webkit-align-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -webkit-flex: 0 0 auto;
}

.css-rbl5we,
[data-css-rbl5we] {
  display: block;
  font-family: "Open Sans", Helvetica, Arial, sans-serif;
  font-weight: 600;
  font-size: 14px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #333333;
}

.css-4p3ux5,
[data-css-4p3ux5] {
  box-sizing: border-box;
  flex: 0 0 auto;
  -webkit-flex: 0 0 auto;
}

.css-1wp5mix,
[data-css-1wp5mix] {
  box-sizing: border-box;
  align-content: center;
  align-items: center;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: row;
  justify-content: center;
  flex: 0 0 auto;
  -webkit-align-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-flex: 0 0 auto;
}

<div
  data-css-iar4wy=""
  data-css-tziwal=""
>
  <div
    data-css-4bx01r=""
    data-css-f86ip=""
    onClick={[Function]}
  >
    <div
      data-css-1eav4r6=""
      data-css-h7582n=""
    >
      <div
        data-css-4p3ux5=""
        data-css-rbl5we=""
        style={
          Object {
            "overflow": "hidden",
            "textOverflow": "ellipsis",
            "whiteSpace": "nowrap",
            "width": "100%",
          }
        }
      >
        string
      </div>
    </div>
    <div
      data-css-1di7qz=""
      data-css-1wp5mix=""
    >
      <div
        data-css-1l7pri3=""
        data-css-4p3ux5=""
        onKeyPress={[Function]}
        tabIndex={1}
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
          data-css-4p3ux5=""
          size={10}
          style={
            Object {
              "fill": "lightGreyIntense",
              "height": 10,
              "stroke": false,
              "width": 10,
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    data-css-4p3ux5=""
    data-css-dbct2k=""
  >
    <p>
      Test child
    </p>
  </div>
</div>
`;
